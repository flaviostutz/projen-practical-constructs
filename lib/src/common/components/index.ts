export * from './base-tooling';
export * from './common-target-type';
export * from './common-targets';
export * from './release-tasks';
export * from './publish';
