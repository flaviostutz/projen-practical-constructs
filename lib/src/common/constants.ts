export const PROJEN_VERSION = '0.91.6';
export const NODE_VERSION = '20.16.0';
