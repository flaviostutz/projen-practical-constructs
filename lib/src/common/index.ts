export * from './components';
export * from './files';
export * from './types';
