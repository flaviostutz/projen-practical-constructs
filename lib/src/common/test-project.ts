import { mkdtempSync } from 'node:fs';
import { tmpdir } from 'node:os';
// eslint-disable-next-line unicorn/import-style
import { join } from 'node:path';

import { Project } from 'projen';

export class TestProject extends Project {
  constructor() {
    const outdir = mkdtempSync(join(tmpdir(), 'test-'));
    super({ name: 'test', outdir });
  }
}
