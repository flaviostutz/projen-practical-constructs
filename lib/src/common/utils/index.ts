export * from './ts2jsii';
export * from './monotag';
