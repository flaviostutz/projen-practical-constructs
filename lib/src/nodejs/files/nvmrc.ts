import { FileBase, IResolver, Project } from 'projen';

export class NvmRcFile extends FileBase {
  private readonly opts: NvmRcOptions;

  constructor(project: Project, opts: NvmRcOptions) {
    super(project, '.nvmrc');
    this.opts = opts;
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  protected synthesizeContent(_resolver: IResolver): string | undefined {
    return `${this.opts.nodeVersion}`;
  }
}

export interface NvmRcOptions {
  /**
   * The version of Python to be added to the file
   * @required
   */
  readonly nodeVersion: string;
}
