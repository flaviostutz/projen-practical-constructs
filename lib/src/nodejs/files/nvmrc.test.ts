/* eslint-disable no-new */
import { Testing } from 'projen';

import { TestProject } from '../../common/test-project';

import { NvmRcFile } from './nvmrc';

describe('NvmRcFile', () => {
  it('synthesizes .nvmrc correctly', () => {
    const prj = new TestProject();
    new NvmRcFile(prj, { nodeVersion: '20.16.0' });
    const out = Testing.synth(prj);
    expect(out['.nvmrc']).toMatchSnapshot();
  });
});
