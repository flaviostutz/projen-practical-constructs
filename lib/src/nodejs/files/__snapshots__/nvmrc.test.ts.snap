// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NvmRcFile synthesizes .nvmrc correctly 1`] = `"20.16.0"`;
